### PR TITLE
feat: use Bugsnag for error monitoring

### PR DESCRIPTION
**Pull request by BrickheadJohnny**
I only added the error handler to our top level error boundary for testing. If we decide to use this tool, we should probably migrate the existing posthog error events to bugsnag too!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,8 +81,9 @@
     "slugify": "^1.6.5",
     "swr": "^2.2.4",
     "uuidv7": "^0.6.3",
-    "viem": "^2.9.6",
-    "wagmi": "^2.5.17",
+    "viem": "2.9.6",
```
Comment (BrickheadJohnny) [inline package.json:84]: encountered a type error in newer versions, so I fixed the versions for these packages

File: package.json [hunk 2]
```diff
@@ -80,8 +81,9 @@
...
+    "wagmi": "2.5.17",
+    "webpack-bugsnag-plugins": "^1.8.0",
```
Comment (cs-balazs) [inline package.json:86]: I'm wondering if this can be a `devDependency` 🤔

File: src/bugsnag.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    window.location.host !== "guild.xyz"
+  )
+    return
+
+  Bugsnag.start({
+    apiKey: process.env.NEXT_PUBLIC_BUGSNAG_KEY,
+    plugins: [],
+    endpoints: {
+      notify: "/api/bugsnag/notify",
+      sessions: "/api/bugsnag/sessions",
```
Comment (cs-balazs) [inline src/bugsnag.ts:16]: Don't we need rewrite rules for these?

**Review by cs-balazs — COMMENTED** [review]
Nice, also, we shouldn't forget to add a domain whitelist on the BugSnag dashboard :ayy:

File: src/bugsnag.ts (hunk 3 above)
Comment (cs-balazs) [reply src/bugsnag.ts:16]: Nvm, found them 😄

**Review by cs-balazs — APPROVED** [review]
Great! I can't wait to see those error events 💯

File: package.json (hunk 2 above)
Comment (BrickheadJohnny) [reply package.json:86]: Good point, seems like it'll work that way too, so reinstalled it as a `devDependency`!

**Pull request merged**